python/cudf/dataframe/series.py

```diff
@@ -1833,6 +1833,14 @@ def quantile(self, q=0.5, interpolation='linear', exact=True,
         DataFrame
 
         """
+
+        if isinstance(q, (int, float)) and not (0 <= q <= 1):
+            raise ValueError("percentiles should all \
+                             be in the interval [0, 1]")
+
```

Comment:
We don't seem to be having a range check for q. Doing it now.
But should we do the same check for all values of `q` if q is a list? Raising this as pandas does this check for all values of q if q is a list.

Reply:
Yes, we should be able to vectorize the operation on the CPU using numpy to accomplish it.